modules/common/src/DataResource.php

```diff
@@ -472,4 +472,11 @@ public function generateChecksum() {
     }
   }
 
+  /**
+   * Return current Unix timestamp.
+   */
+  public function getCurrentTime() {
```

Comment:
This is only used for the database operations in PostImportResult and doesn't make much sense on the DataResource class. I'd say let's make current time be an argument for PostImportResult's constructor, then there's a few ways you could mock or set it in the tests to make sure a predictable value gets passed.

Reply:
I originally had this method as a public method in the PostImportResult class. I had issues getting mocking it before the factory class initializes the PostImportResult object. I can change this to use Drupal\Component\Datetime\Time class in the PostImportResult and PostImportResultFactory. I can then mock getCurrentTime method from the Time class. That should work in theory. How does that sound?